hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/commit/BaseCommitActionExecutor.java

```diff
@@ -235,11 +235,15 @@ protected void commit(HoodieWriteMetadata<O> result, List<HoodieWriteStat> write
       LOG.info("Committed " + instantTime);
       result.setCommitMetadata(Option.of(metadata));
       // update cols to Index as applicable
-      HoodieColumnStatsIndexUtils.updateColsToIndex(table, config, metadata,
-          (Functions.Function2<HoodieTableMetaClient, List<String>, Void>) (metaClient, columnsToIndex) -> {
-            updateColumnsToIndexForColumnStats(metaClient, columnsToIndex);
-            return null;
-          });
+      try {
+        HoodieColumnStatsIndexUtils.updateColsToIndex(table, config, metadata,
+                (Functions.Function2<HoodieTableMetaClient, List<String>, Void>) (metaClient, columnsToIndex) -> {
+                  updateColumnsToIndexForColumnStats(metaClient, columnsToIndex);
+                  return null;
+                });
+      } catch (UnsupportedOperationException uoe) {
```

Comment:
Could we catch a specific exception for bootstrap here?

Reply:
Have you checked that there is no failed deltacommit in MDT this case and files partition is intact, and the index ready to use does not contain col_stats in the table config?

Reply:
Would be good to add this validation in the test.